### PR TITLE
Feature: Added Treeview Component

### PR DESCRIPTION
**Pull request by hgeldenhuys**
Just check it out and give me input. I'm still trying to find my feet in your repo, how it works and style-guide etc

### CODE REVIEW

File: src/mantine-core/src/components/Treeview/interfaces.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { CSSProperties, ReactNode } from 'react';
```
Comment (rtivital) [inline src/mantine-core/src/components/Treeview/interfaces.tsx:1]: Rename `interfaces.tsx` -> `types.tsx`

File: src/mantine-core/src/components/Treeview/utils.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export function uuidv4() {
```
Comment (rtivital) [inline src/mantine-core/src/components/Treeview/utils.ts:1]: We already have a function to generate uuid – you can import it:

```tsx
import { randomId } from '@mantine/hooks';
```

File: src/mantine-core/src/components/Treeview/styles.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { createUseStyles } from 'react-jss';
```
Comment (rtivital) [inline src/mantine-core/src/components/Treeview/styles.tsx:1]: Rename `styles.tsx` -> `Tree.styles.ts`

**Review by rtivital — COMMENTED** [review]
Thanks for Pull Request, this is still WIP and on early stages so I did not add reviews to the whole code, just the most basic things. Here are some extra points:
- Rename whole component to just a `Tree`
- Replace `items` prop with children, this approach will make component more extendable

Here is an example API for children:

```tsx
<Tree>
  <Leaf label="index.js" />
  <Leaf label={() => <TextInput placeholder="Edit file name" variant="unstyled" />} />
  <Leaf label="folder">
    <Leaf label="file-in-folder.js" />
    <Leaf label="file-in-folder-2.js" />
    <Leaf label="folder-in-folder">
      <Leaf label="file-in-folder.js" />
      <Leaf label="file-in-folder.js" />
    </Leaf>
  </Leaf>
</Tree>
```

**Comment by MrHBS** [comment]
Any chance we get this going?

**Comment by rtivital** [comment]
Tree component will be added as part of one of 3.x releases

**Pull request closed without merging**